packages/web3-eth-ens/src/resolver.ts

```diff
@@ -0,0 +1,114 @@
+import { Address, sha3, isHexStrict } from 'web3-utils';
+import { Contract, NonPayableCallOptions } from 'web3-eth-contract';
+import { inputAddressFormatter, ResolverMethodMissingError } from 'web3-common';
+import { interfaceIds, methodsInInterface } from './config';
+import { Registry } from './registry';
+import { RESOLVER } from './abi/resolver';
+import { namehash } from './utils';
+
+//  Default public resolver
+//  https://github.com/ensdomains/resolvers/blob/master/contracts/PublicResolver.sol
+
+export class Resolver {
+	private readonly registry: Registry;
+
+	public constructor(registry: Registry) {
+		this.registry = registry;
+	}
+
+	private async getResolverContractAdapter(ENSName: string) {
+		//  TODO : (Future 4.1.0 TDB) cache resolver contract if frequently queried same ENS name, refresh cache based on TTL and usage, also limit cache size, optional cache with a flag
+		return this.registry.getResolver(ENSName);
+	}
+
+	//  https://eips.ethereum.org/EIPS/eip-165
+	// eslint-disable-next-line class-methods-use-this
+	public async checkInterfaceSupport(
```

Comment:
This could be a static method?

Reply:
I dont think so, it calls the resolverContract on line 38-40

Reply:
I dont think this matters too much, resolver is only used internally by us and is only exposed through the ENS class